lib/utils.js

```diff
@@ -234,19 +241,21 @@ exports.parseJSON = parseJSON;
  */
 var extractMetadataFromHeaders = function extractMetadataFromHeaders(response) {
   var metadata = {};
-  var rawHeaderCounter = 0;
-  for(var field in response.headers) {
-    rawHeaderCounter++;
-    if (/x-ms-meta-/.test(field)) {
-      // Metadata names must adhere to the naming rules for C# identifiers, which are case-insensitive,
-      // meaning that you can set,for example, a metadata with the following form:
-      // {
-      //    applicationName: 'fast-azure-blob-storage'
-      // }
-      // In order to return the original metadata name, the header names should be read from response.rawHeaders.
-      // That is because 'https' library returns the response headers with lowercase.
-      var key = response.rawHeaders[rawHeaderCounter * 2 - 2];
-      metadata[key.substr(10)] = response.headers[field];
+  /*
+   * Metadata names must adhere to the naming rules for C# identifiers, which are case-insensitive,
```

Comment:
You mean, case-insensitive case-preserving, allowed strings are any valid C# identifier that can be used in a HTTP header.